### ISSUE TITLE
Bug: Getting `TypeError: Nested 'extends' is not allowed.`

### ISSUE THREAD

**Issue by vbeffa** [typescript-eslint/typescript-eslint#11742]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

I am trying to update my project to the latest version of `eslint-plugin-react-hooks` and in doing so had to redo my entire eslint.config.js. See config below. I stripped everything out to the minimum to still reproduce the error. When I run the linter, I get the error shown in `Actual Result`. I've searched for it and can't find another instance of it.

### ESLint Config

```javascript
import { defineConfig } from 'eslint/config';
import tseslint from '@typescript-eslint/eslint-plugin';

export default defineConfig(
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js'],
    extends: [
      tseslint.configs.recommended,
    ]
  }
);
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I don't have a nested extends, so I would expect the linter to work with my config.

### Actual Result

```
❯ npm run lint

> vite-project@0.0.0 lint
> eslint .


Oops! Something went wrong! :(

ESLint: 9.39.1

TypeError: Nested 'extends' is not allowed.
    at processExtends (./node_modules/@eslint/config-helpers/dist/cjs/index.cjs:462:10)
    at ./node_modules/@eslint/config-helpers/dist/cjs/index.cjs:504:38
    at Array.flatMap (<anonymous>)
    at processConfigList (./node_modules/@eslint/config-helpers/dist/cjs/index.cjs:504:20)
    at defineConfig (./node_modules/@eslint/config-helpers/dist/cjs/index.cjs:540:9)
    at file://./eslint.config.js?mtime=1762380747314:5:16
    at ModuleJob.run (node:internal/modules/esm/module_job:371:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:669:26)
    at async dynamicImportConfig (./node_modules/eslint/lib/config/config-loader.js:186:17)
    at async loadConfigFile (./node_modules/eslint/lib/config/config-loader.js:276:9)
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `TypeScript`                           | `5.9.3` |
| `ESLint`                               | `9.39.1` |
| `node`                                 | `24.6.0` |

**Comment by vbeffa** [comment]
It also appears that a directive like

```ts
    extends: [
      reactHooks.configs['recommended-latest']
    ],
```

is not recognized, as I get the error

```

ESLint: 9.39.1


A config object has a "plugins" key defined as an array of strings. It looks something like this:

    {
        "plugins": ["react-hooks"]
    }

Flat config requires "plugins" to be an object, like this:

    {
        plugins: {
            react-hooks: pluginObject
        }
    }
```

**Comment by vbeffa** [comment]
I can, however, successfully run the linter with

```ts
import reactRefresh from 'eslint-plugin-react-refresh';

...

    extends: [
      reactRefresh.configs.vite
    ],
...
```

**Comment by bradzacher** [comment]
https://typescript-eslint.io/getting-started/

You're importing our config from the wrong package.
Please review our getting started guide for details on how to correctly setup your config.
